rcsb/utils/dictionary/DictMethodEntityHelper.py

```diff
@@ -97,13 +98,26 @@ def __processSiftsAlignments(self, dataContainer):
         #
         # Process sifts alignments -
         siftsAlignD = {}
+        asymMaxAlignLengthD = {}
         for asymId, authAsymId in asymAuthIdD.items():
             if instTypeD[asymId] not in ["polymer", "branched"]:
                 continue
             entityId = asymIdD[asymId]
-            # accumulate the sifts alignments by entity.
+            #
+            asymMaxAlignLength = asymMaxAlignLengthD.get((entryId, entityId), 0)
+            asymSeqAlignObjL = self.__ssP.getSeqAlignObjList(entryId, authAsymId)
+            asaolenLength = sum([seqAlignObj.getEntityAlignLength() for seqAlignObj in asymSeqAlignObjL])
+            logger.debug("asaolenLength %r for list: %r", asaolenLength, asymSeqAlignObjL)
+            #
+            # accumulate only the longest sifts alignments by entity.
+            # Only keep the chain with the longest alignment
+            if asaolenLength > asymMaxAlignLength:
+                siftsAlignD[(entryId, entityId)] = asymSeqAlignObjL
+                logger.debug("siftsAlignD: %r", siftsAlignD)
+                asymMaxAlignLengthD[(entryId, entityId)] = asaolenLength
+            #
             # siftsAlignD.setdefault((entryId, entityId), []).extend([SeqAlign("SIFTS", **sa) for sa in self.__ssP.getIdentifiers(entryId, authAsymId, idType="UNPAL")])
-            siftsAlignD.setdefault((entryId, entityId), []).extend(self.__ssP.getSeqAlignObjList(entryId, authAsymId))
+
         for (entryId, entityId), seqAlignObjL in siftsAlignD.items():
```

Comment:
With the proposed changes, the code segment in lines 129-136 now seems to be unnecessary based on the new additions in lines 107-117. 

If I understand this correctly (and please correct me if I am wrong), identifying the longest overlapping entity region only applies if there is more than one alignment between an entity and a reference sequence, which happens with homomeric structures. But that logic is now addressed earlier when only the longest alignment for an entity is selected. So the code segment (lines 129-136) seems redundant.

Reply:
Thank you Brinda, you are correct that was redundant code. It was basically rewrangling the data into the a new dictionary and in the same structure. I've consolidated the code and just finished testing it over a full reload, which appeared to work for the `CoreDataLoading` task (as the alignment was then fixed).

However, upon running the next task of the workflow, `ExdbDataLoading`, I discovered that task ends up overwriting the alignments by the [ReferenceSequenceAssignmentAdapter.py](https://github.com/rcsb/py-rcsb_exdb/blob/556f3491c8df14c45059218fcf8e1a86df58c3f4/rcsb/exdb/seq/ReferenceSequenceAssignmentAdapter.py#L311C20-L311C20) process. So now, the alignments are reverted back to the incorrect values. Accordingly, I'll need to update the code in the py-rcsb_exdb package to resolve that.

Reply:
OK, I've implemented and tested a fix here: https://github.com/rcsb/py-rcsb_utils_seq/pull/15

Reply:
Thanks @piehld. Looks good to me.